src/install_pypi/plan/cache.rs

```
@@ -0,0 +1,233 @@
+                    }
+                    if entry.dist.filename == wheel.best_wheel().filename {
+                        Some(&entry.dist)
+                    } else {
+                        None
+                    }
+                });
+
+                if let Some(distribution) = cached {
+                    Ok(Some(CachedDist::Registry(distribution.clone())))
+                } else {
+                    Ok(None)
+                }
+            }
+            Dist::Built(BuiltDist::DirectUrl(wheel)) => {
+                // Find the exact wheel from the cache, since we know the filename in advance.
+                let cache_entry = uv_cache
+                    .shard(
+                        CacheBucket::Wheels,
+                        WheelCache::Url(&wheel.url).wheel_dir(wheel.name().as_ref()),
+                    )
+                    .entry(format!("{}.http", wheel.filename.cache_key()));
+
+                // Read the HTTP pointer.
+                match HttpArchivePointer::read_from(&cache_entry) {
+                    Ok(Some(pointer)) => {
+                        let cache_info = pointer.to_cache_info();
+                        let archive = pointer.into_archive();
+                        let cached_dist = CachedDirectUrlDist {
+                            filename: wheel.filename.clone(),
+                            url: VerbatimParsedUrl {
+                                parsed_url: wheel.parsed_url(),
+                                verbatim: wheel.url.clone(),
+                            },
+                            hashes: archive.hashes,
+                            cache_info,
+                            path: uv_cache.archive(&archive.id).into_boxed_path(),
+                        };
+
+                        Ok(Some(CachedDist::Url(cached_dist)))
+                    }
+                    Ok(None) => Ok(None),
+                    Err(err) => {
+                        tracing::debug!(
```

should we raise an error here?